### PR TITLE
PLAD-112

### PR DESCRIPTION
**Pull request by BassamZ**

### CODE REVIEW

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _createnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Student name',
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:105]: From Figma, this should be just "Name"

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        return null;
+      },
+    );
+  }
+
+Widget _createsurnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Student surname',
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:125]: From Figma, this should be just "Surname"

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student email';
+        }
+        return null;
+      },
+    );
+  }
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:180]: Why is this here? It should not be implemented in this screen

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                      vertical: 5.0,
+                    ),
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        _createnameField(student),
+                        _createsurnameField(student),
+                        _createemailField(student),
+                        _createpasswordField(student),
+                        _createDescriptionField(student),
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:71]: The Phone number field is missing from the screen

File: frontend/lib/screens/new_student/new_student.dart [hunk 5]
```diff
@@ -11,7 +12,8 @@ class NewStudent extends StatefulWidget {
 
 class _NewStudentState extends State<NewStudent> {
   bool _firstStep = true;
-  Student _newStudent = Student(name: "Bassam", surname: "Zabad");
+  //Student _newStudent = Student(name: "Bassam", surname: "Zabad");
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/new_student.dart:15]: We can delete this line if we don't think it will ever been used again

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 3 above)
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:180]: Why is this here? It should not be implemented in this screen

**Review by Fabs97 — CHANGES_REQUESTED** [review]

**Review by aidadenisa — APPROVED** [review]

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 4 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart:71]: there is no phone number field in the database so I did not add it

Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart:71]: I will send to Anna, so we can add that

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 1 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart]: Done

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 2 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart]: Done

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 4 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart:70]: Done. the frontend is corrected, but the backend and database need to be edited.

File: frontend/lib/screens/new_student/new_student.dart (hunk 5 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/new_student.dart]: Done

File: frontend/lib/screens/new_student/local_widget/student_form.dart (hunk 3 above)
Comment (BassamZ) [reply frontend/lib/screens/new_student/local_widget/student_form.dart]: Done

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+                      vertical: 5.0,
+                    ),
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        _createnameField(student),
+                        _createsurnameField(student),
+                        _createemailField(student),
+                        _createDescriptionField(student),
+                        _createphoneField(student),
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:71]: @BassamZ please stick to the design, this way it is not correct: phone field must be before the description field

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a phone number';
+        }
+        return null;
+      },
+    );
+  }
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:205]: @BassamZ it would be great if this field would take only numbers as an input. You can check an example in the creation of the placement, with the salary or the hours per week field

File: frontend/lib/screens/new_student/local_widget/student_form.dart [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+        setState(() {
+          student.email = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student email';
+        }
+        return null;
+      },
```
Comment (Fabs97) [inline frontend/lib/screens/new_student/local_widget/student_form.dart:158]: @BassamZ would it be possible to add an email validation with a regex? Something like this should work: 
^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$

Comment (Fabs97) [reply frontend/lib/screens/new_student/local_widget/student_form.dart:158]: You can check how to do a regex validation in Flutter [here](https://stackoverflow.com/questions/56390839/flutter-regex-in-textformfield)

**Review by Fabs97 — CHANGES_REQUESTED** [review]

**Review by aidadenisa — APPROVED** [review]

**Review by Fabs97 — APPROVED** [review]
